src/classes/CSVReader.cls

```diff
@@ -0,0 +1,216 @@
+*
+*/
+public with sharing class CSVReader {
+    
+    public static String lineTerminator = ConstantUtil.CRLF;
+    public static String valueSeparator = ConstantUtil.COMMA;
+    
+    public class CSVReadException extends Exception {}
+    
+    // return parsed file as "matrix" - rows represent complete records, and columns are fields
+    
+    public static List<List<String>> readCSVFile(Blob fileContent)
+    {
+        String fileString = fileContent.toString();
+        fileContent = null;
+        return readCSVFile(fileString);
+    }
+    
+    public static List<List<String>> readCSVFile(String fileString)
+    {
+        //fileString = fileString.replaceAll('\\s*\\z', '');    // remove any trailing empty lines
+        if (!fileString.endsWith(lineTerminator))   // parser expects all rows to have a endline as a terminator
+        {
+            fileString = fileString + lineTerminator;
+        }
+        
+        List<List<String>> objectValues = new List<List<String>>();
+        List<String> fieldValues = new List<String>();
+        
+        Boolean eod = false;  // eod = End Of Document
+        while(!eod)
+        {
+            CSVValue csvVal = readCSVValue(fileString);
+            fieldValues.add(csvVal.value);
+            //check if the field value is null or blank
+            if (csvVal.value=='' || csvVal.value==null)
+            {
+            	//trow an error if there are any groups or members left blank(groups and members are required for upload)
+            	throw new CSVReadException(ConstantUtil.CSV_CONTAINS_NULL_FIELDS);
+            }
+            if(csvVal.terminator == lineTerminator)
+            {
+                // if row was not blank - contains one or more non-blank value
+                if(String.isNotBlank(fieldValues.get(0)) || fieldValues.size() > 1)
+                {
+                    objectValues.add(fieldValues);
+                
+                    // check that each row has the same number of columns
+                    if(objectValues.get(0).size() != fieldValues.size())
+                    {
+                        throw new CSVReadException(
+                            'Mismatch in number of columns.\n'
+                            + 'Expected: ' + objectValues.get(0).size() + '(1. row)\n'
+                            + 'Actual: ' + fieldValues.size() + '(' + objectValues.size() + '. row)\n');
+                    }
+                }
+                
+                fieldValues = new List<String>();
+            }
+            
+            if(csvVal.biteSize() == fileString.length() || fileString.length() == 0)
+            {
+                eod = true;
+            }
+            else
+            {
+                fileString = fileString.substring(csvVal.biteSize());
+            }
+        }
+        
+        return objectValues;
+    }
+    
+    /**/
+    
+    // read single (field) value from the file
+    public static CSVValue readCSVValue(String strData)
```

Comment:
Methods should use the `lowerCamelCase` naming convention. I suggest rename accordingly.